cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -761,6 +762,64 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
+     *
+     * @param configuration The Kafka configuration. Needed to check the min.insync.replica and CC metrics topic settings.
+     * @param result The KafkaCluster object being contructed. Needed for default settings.
+     * @param kafkaClusterSpec The KafkaClusterSpec objecct needed for the checking the number of brokers in the cluster.
+     * @return The recommended CC metrics topic replication factor as a string.
+     */
+    private static String checkCCMetricsReplication(KafkaConfiguration configuration, KafkaCluster result, KafkaClusterSpec kafkaClusterSpec) {
+
+        int configuredCCReplicationFactor;
+        if (configuration.getConfigOption(CC_REPLICATION_FACTOR_CONFIG_FIELD) == null) {
+            configuredCCReplicationFactor = Integer.parseInt(result.kafkaDefaultReplicationFactor);
+        } else {
+            configuredCCReplicationFactor = Integer.parseInt(configuration.getConfigOption(CC_REPLICATION_FACTOR_CONFIG_FIELD));
+        }
+        int brokerMinInsyncReplicas;
+        if (configuration.getConfigOption(KAFKA_MIN_INSYNC_REPLICA_CONFIG_FIELD) == null) {
+            brokerMinInsyncReplicas = Integer.parseInt(result.kafkaDefaultMinInsyncReplias);
+        } else {
+            brokerMinInsyncReplicas = Integer.parseInt(configuration.getConfigOption(KAFKA_MIN_INSYNC_REPLICA_CONFIG_FIELD));
+        }
+
+        // We want the CC metrics topics to have a replication factor at least one higher than the broker minimum so that metrics
+        // producer acks are likely to be acknowledged faster.
+        int checkedCcReplicationFactor = Math.max(configuredCCReplicationFactor, brokerMinInsyncReplicas + 1);
+
+        // However, we need to check that we are not setting the replication factor to more than the available number of brokers.
+        if (checkedCcReplicationFactor > kafkaClusterSpec.getReplicas()) {
+            if (kafkaClusterSpec.getReplicas() >= brokerMinInsyncReplicas) {
+                // If the number of brokers is greater than or equal to the cluster minISR then set the CC metric topic replicas to
+                // equal the number of brokers.
+                checkedCcReplicationFactor = kafkaClusterSpec.getReplicas();
+            } else {
+                // If we are here then we have issue because minISR has been set higher than the number of brokers, which is a bad idea.
+                // We are assuming the user must be ok with this so we set the CC
+                checkedCcReplicationFactor = brokerMinInsyncReplicas;
+                log.warn("The configured {} ({}) is less than the number of brokers in the cluster ({}). This is not recommended, " +
+                         "either add additional brokers or reduce the {} to be less than or equal to the number of brokers.",
+                         KAFKA_MIN_INSYNC_REPLICA_CONFIG_FIELD, brokerMinInsyncReplicas, kafkaClusterSpec.getReplicas(),
+                         KAFKA_MIN_INSYNC_REPLICA_CONFIG_FIELD);
```

Comment:
I think it could depend case by case by the user. It's a tradeoff between availability and consistency. If you have min ISR less broker replicas you can lose consistency but gaining availability (because producer can still produce even if a broker goes down). If you force min ISR = broker replicas you have less availability but forcing consistency.